core/src/main/java/org/elasticsearch/action/admin/cluster/shards/TransportClusterSearchShardsAction.java

```diff
@@ -67,12 +76,20 @@ protected ClusterSearchShardsResponse newResponse() {
     }
 
     @Override
-    protected void masterOperation(final ClusterSearchShardsRequest request, final ClusterState state, final ActionListener<ClusterSearchShardsResponse> listener) {
+    protected void masterOperation(final ClusterSearchShardsRequest request, final ClusterState state,
+                                   final ActionListener<ClusterSearchShardsResponse> listener) {
         ClusterState clusterState = clusterService.state();
         String[] concreteIndices = indexNameExpressionResolver.concreteIndexNames(clusterState, request);
         Map<String, Set<String>> routingMap = indexNameExpressionResolver.resolveSearchRouting(state, request.routing(), request.indices());
+        Map<String, AliasFilter> indicesAndFilters = new HashMap<>();
+        for (String index : concreteIndices) {
+            AliasFilter aliasFilter = indicesService.buildAliasFilter(clusterState, index, request.indices());
```

Comment:
I think that the AliasFilter can never be null? Rather the inner query builder can be. So I thought it is good to always add the alias filter object anyways, so there is something in the map. The object has been created anyways. Don't we do the same in TransportSearchAction?